addons/project-title-generator/userscript.js

```diff
@@ -0,0 +1,124 @@
+      // A file upload will then dispatch DONE_LOADING_VM_WITHOUT_ID, but we should ignore it
+      isFileUpload = true;
+    }
+  });
+
+  // Create the randomizer button
+  let button;
+  createButton();
+
+  addon.tab.redux.addEventListener("statechanged", async (e) => {
+    if (e.detail.action.type === "projectTitle/SET_PROJECT_TITLE") {
+      let showButton = await shouldButtonShow();
+      if (showButton) createButton();
+      else removeButton();
+    }
+  });
+  addon.tab.addEventListener("urlChange", () => {
+    if (!addon.self.disabled && addon.tab.editorMode === "editor") createButton();
+    else if (addon.tab.editorMode !== "editor" && button) removeButton();
+  });
+  addon.self.addEventListener("disabled", () => removeButton());
+  addon.self.addEventListener("reenabled", () => createButton());
+  addon.settings.addEventListener("change", async () => {
+    let showButton = await shouldButtonShow();
+    if (showButton) createButton();
+    else removeButton();
+  });
+
+  async function shouldButtonShow() {
+    let currentName = await addon.tab.redux.state.scratchGui.projectTitle;
+    if (!addon.settings.get("only-untitled") || currentName === "" || currentName.includes("Untitled")) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  async function createButton() {
+    let showButton = await shouldButtonShow();
+    if (!showButton || document.querySelector("#sa-project-title-generator-button") !== null) {
+      return;
+    }
+    let nameContainer = await addon.tab.waitForElement('[class*="menu-bar_menu-bar-item"][class*="menu-bar_growable"]');
+    nameContainer.classList.add("sa-project-title-generator");
+    let nameField = await addon.tab.waitForElement('[class*="project-title-input_title-field"]');
+    if (button) button.remove();
+    button = document.createElement("span");
+    button.id = "sa-project-title-generator-button";
+    button.className = addon.tab.scratchClass(
+      "button_outlined-button",
+      "menu-bar_menu-bar-button",
+      "community-button_community-button"
+    );
+    let buttonImg = document.createElement("img");
+    buttonImg.id = "sa-project-title-generator-button-img";
+    buttonImg.className = addon.tab.scratchClass("community-button_community-button-icon", "button_icon");
+    buttonImg.src = addon.self.dir + "/dice-five.svg";
+    button.appendChild(buttonImg);
+    nameField.after(button);
+    button.addEventListener("click", () => setProjectName());
+  }
+
+  async function removeButton() {
+    let nameContainer = await addon.tab.waitForElement('[class*="menu-bar_menu-bar-item"][class*="menu-bar_growable"]');
+    nameContainer.classList.remove("sa-project-title-generator");
+    button.remove();
+  }
+
+  async function setProjectName() {
+    let newName =
+      msg("adj-" + randi(ADJ_COUNT)) + " " + msg("adj-" + randi(ADJ_COUNT)) + " " + msg("noun-" + randi(NOUN_COUNT));
```

Comment:
In some languages, Spanish for example, the adjectives go after the noun.

Reply:
True. Or Romanian. But I think this won't be translated.

Reply:
True, I forgot about that. I'll adjust the translation string to support this.